scripts/linters/find_banned_deps.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'toml-rb'
+
+whitelist = {
+  # Check all `Cargo.toml` files for `time` crate
+  ['time'] => []
+  # The following code will be uncommented as part of `PR#2880: https://github.com/ChainSafe/forest/pull/2880`
+  # # Create exceptions for `forest_shim` and `forest_interpreter` (only places `fvm` & `fvm3` are allowed).
+  # %w[fvm fvm3] => ['utils/forest_shim/Cargo.toml', 'vm/interpreter/Cargo.toml']
+}
+
+# Iterate over all TOML files in the repository
+Dir.glob('**/*.toml').each do |file|
+  toml = TomlRB.load_file(file)
+
+  toml['dependencies']&.each do |dep|
+    whitelist.each_pair do |crates, exceptions|
+      crates.each do |crate|
+        if dep[0] == crate && !(exceptions.include? file)
+          puts "[#{dep[0]}] is being used in [#{file}], but this crate is banned!"
```

Comment:
This won't ever be caught in the CI, no? Except for an error in the script itself. I mean, if someone were to use a banned dependency, the script would still exit with a success code. Is it a desired behaviour?

Reply:
Nope, good point. Fixed this.